src/algoliasearch.helper.js

```diff
@@ -886,6 +889,12 @@ AlgoliaSearchHelper.prototype.setIndex = function(name) {
  * helper.setQueryParameter('hitsPerPage', 20).search();
  */
 AlgoliaSearchHelper.prototype.setQueryParameter = function(parameter, value) {
```

Comment:
it's not the right place here, people can also call setQueryParameters, but both this and setQueryParameters use setState

Reply:
oh right, so if I understand correctly, everything leads to here: https://github.com/algolia/algoliasearch-helper-js/blob/fix/invalid-usertoken/src/SearchParameters/index.js#L1369:L1369

Reply:
@Haroenv 
![image](https://user-images.githubusercontent.com/499898/101041978-f2c2c680-357c-11eb-9c25-c114b2f0df69.png)
https://excalidraw.com/#json=4883557457592320,c67cypVCpL_-WxoGxVuwKQ

I drew the flow where state can change except for facets (because facet-related methods can never update userToken).

I think everything goes into the constructor of SearchParameters.

Reply:
yep, the warning should only be in the constructor